src/components/modalWindows/addMovie/addMovie.js

```diff
@@ -1,151 +1,193 @@
-      event.preventDefault()
-      const form = event.target;
-      if(!editMovie){
-        const newMovie = {
-          id: `MO${new Date().valueOf()}VIE`,
-          title: form[0].value,
-          data: form[1].value,
-          url: form[2].value,
-          genre: form[3].value,
-          overview: form[4].value,
-          runtime: form[5].value,
-        };
-        // -----TEST FORM ADDMOVIE SUBMIT-----
-        console.log('TEST FORM ADDMOVIE SUBMIT', newMovie)
-        // --------------------------
-      } else {
-        const editMovie = {
-          id: props.id,
-          title: form[0].value,
-          data: form[1].value,
-          url: form[2].value,
-          genre: form[3].value,
-          overview: form[4].value,
-          runtime: form[5].value,
-        };
-        // -----TEST FORM EDITMOVIE SUBMIT-----
-        console.log('TEST FORM EDITMOVIE SUBMIT', editMovie)
-        // --------------------------
-      }
 
-      handleClose();
-    }, [])
+    const formik = useFormik({
+        initialValues: {
+          ...movie
+        },
+        validationSchema: Yup.object({
+          title: Yup.string()
+            .required('Required'),
+          release_date: Yup.string()
+            .max(10, 'Must be 8 characters')
+            .required('Required'),
+          genres: Yup.string()
+            .max(100, 'Must be 8 characters')
+            .required('Required'),
+          poster_path: Yup.string().url().required('Wrong URL'),
+          overview: Yup.string()
+            .max(500, 'Must be 15 characters or less')
+            .required('Required'),
+          runtime: Yup.number()
+            .positive()
+            .integer()
+            .required('Required'),
+        }),
+        onSubmit: values => {
+          if(!editMovie){
+                const newMovie = {
+                  ...values,
+                  genres: values.genres.split(','),
+                  tagline: values.title,
+                  vote_average: 5.5,
+                  vote_count: 100,
+                  budget: 55000000,
+                  revenue: 136906000,
+                };
+                addNewMovie(newMovie);
+              } else {
+                editExistMovie(values);
+              }
+        
+              handleClose();
+        }
+      });
 
     return (
       <form
         role="form"
         className="col-12"
-        onSubmit={handleSignIn}
+        onSubmit={formik.handleSubmit}
       >
-        <h2 className="title">Add movie</h2>
+        <h2 className="title">{editMovie ? 'Edit movie' : 'Add movie'}</h2>
         <div className="form-group">
           {editMovie && (
             <Fragment>
               <label htmlFor="id">Movie ID</label>
               <p 
                 name="id"
                 id="id">
-                {id}
+                {activeId}
               </p>
             </Fragment>
           )}
           <label htmlFor="title">Title</label>
           <input
```

Comment:
https://formik.org/docs/api/field
The best way is to use a wrapper, it isn't necessary. This example from my project:
```javascript
import React, { memo, useCallback } from 'react';
import PropTypes from 'prop-types';
import { useField, useFormikContext } from 'formik';
import { Box, FormHelperText } from '@material-ui/core';
import FormLabel from '../FormLabel';

import useStyles from './formikField.styles';

const isEvent = (event) => event && (event instanceof Event || event.nativeEvent instanceof Event);

const FormikField = ({
  title,
  direction,
  alignItems,
  component: Component,
  name,
  helperText,
  my,
  mb,
  mt,
  width,
  ...props
}) => {
  const { submitCount } = useFormikContext();
  const [field, meta, helper] = useField(name);
  const { helperText: helperTextStyles, secondLine } = useStyles();
  const { setValue, setTouched } = helper;
  const { onChange, onBlur } = field;
  const { error, touched } = meta;

  const handleChange = useCallback(async (eventOrValue) => {
    if (isEvent(eventOrValue)) {
      onChange(eventOrValue);
    } else {
      await setValue(eventOrValue, false);
      setTouched(true, true);
    }
  }, [onChange, setTouched, setValue]);

  const handleBlur = useCallback((eventOrValue) => {
    if (isEvent(eventOrValue)) {
      onBlur(eventOrValue);
    } else {
      setTouched(true);
    }
  }, [onBlur, setTouched]);

  const isError = !!error && (touched || !!submitCount);

  return (
    <Box
      display="flex"
      flexDirection={direction}
      alignItems={alignItems}
      mb={mb || my}
      mt={mt || my}
    >
      { title && (
        <Box mb={direction === 'row' ? 0 : 1} mr={1}>
          <FormLabel title={title} />
        </Box>
      )}
      <Box flexGrow={1} position="relative">
        <Box width={width}>
          <Component
            {...props}
            {...field}
            onChange={handleChange}
            onBlur={handleBlur}
            error={isError || null}
            name={name}
          />
        </Box>
        <FormHelperText
          className={helperTextStyles}
          error={isError && !helperText}
        >
          {!helperText && isError ? error : helperText}
        </FormHelperText>
        {helperText && isError && (
          <FormHelperText
            className={`${helperTextStyles} ${secondLine}`}
            error={isError}
          >
            {error}
          </FormHelperText>
        )}
      </Box>
    </Box>
  );
};

FormikField.defaultProps = {
  direction: 'column',
  alignItems: 'stretch',
  title: '',
  helperText: '',
  my: 3,
  mb: 0,
  mt: 0,
  width: 'auto',
};

FormikField.propTypes = {
  component: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.shape({}),
  ]).isRequired,
  name: PropTypes.string.isRequired,
  direction: PropTypes.string,
  title: PropTypes.string,
  alignItems: PropTypes.string,
  helperText: PropTypes.string,
  my: PropTypes.number,
  mb: PropTypes.number,
  mt: PropTypes.number,
  width: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
};

export default memo(FormikField);

Reply:
OK